### PR TITLE
telink: tl321x: add Tl321x for TLX Platform based on BLE release v4.0.4.2 

### PR DESCRIPTION
**Pull request by damien0x0023**
- add new board tl3218x.
- add tl321x as the first SOC series in the TLX platform.
- update sys_init argument list in TLX soc.c.
- add Zephyr drivers for the TLX platform.
- update hal_telink revision into west.yml.
- update the workflow for tl321x.
- fix compliance issues

### CODE REVIEW

**Comment by andriy-bilynskyy** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/9e86abb71b179007fc9a1fa03e84dd354b77b560 | https://github.com/telink-semi/hal_telink/commit/430c104d68cb63ae23f91c7a1cda6e506822fb45 (develop) | [telink-semi/hal_telink@9e86abb7..430c104d](https://github.com/telink-semi/hal_telink/compare/9e86abb71b179007fc9a1fa03e84dd354b77b560..430c104d68cb63ae23f91c7a1cda6e506822fb45) |
| openthread_telink_lib | https://github.com/telink-semi/openthread_telink_lib/commit/80645ba0b4b589afd8c0712cb1a01091f4b35ab8 | https://github.com/telink-semi/openthread_telink_lib/commit/b5a4ac39d4823a627d8c785879348bf92c49ad7c (main) | [telink-semi/openthread_telink_lib@80645ba0..b5a4ac39](https://github.com/telink-semi/openthread_telink_lib/compare/80645ba0b4b589afd8c0712cb1a01091f4b35ab8..b5a4ac39d4823a627d8c785879348bf92c49ad7c) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

File: drivers/ieee802154/ieee802154_b9x_frame.c
Comment (s07641069) [inline drivers/ieee802154/ieee802154_b9x_frame.c:1]: Why this B9X file changed in case if you created separate ieee802154_tlx_frame.c file ?

**Review by s07641069 — CHANGES_REQUESTED** [review]
PR changes looks ok
Finish HAL PR and after squash Zephyr commits into one as well

**Comment by s07641069** [comment]
@damien0x0023 
Compliance and Documentation Build needs to be fixed

File: drivers/ieee802154/ieee802154_b9x_frame.c
Comment (damien0x0023) [reply drivers/ieee802154/ieee802154_b9x_frame.c:1]: I believe we should not change ieee802154_b9x_frame.c and will revert it.

Comment (damien0x0023) [reply drivers/ieee802154/ieee802154_b9x_frame.c:1]: Done.

File: west.yml [hunk 1]
```diff
@@ -216,7 +216,7 @@ manifest:
       path: modules/lib/open-amp
     - name: openthread_telink_lib
       url: https://github.com/telink-semi/openthread_telink_lib
-      revision: 80645ba0b4b589afd8c0712cb1a01091f4b35ab8
+      revision: 84e12fd24613b2871c47378c39c2f44447f528ba
```
Comment (s07641069) [inline west.yml]: @damien0x0023 this hash is from tl321x_skeleton branch of openthread_telink_lib
Please merge this branch into main and update hash after

**Review by s07641069 — CHANGES_REQUESTED** [review]

File: west.yml (hunk 1 above)
Comment (damien0x0023) [reply west.yml]: I see.
Just create the PR there.
It's better to have @pulin1103 review and confirm before merging.

**Comment by damien0x0023** [comment]
> PR changes looks ok Finish HAL PR and after squash Zephyr commits into one as well

Update hal_telink revision and squash all commits into one.

**Review by s07641069 — APPROVED** [review]

**Pull request merged**